src/main/java/com/treasure/hunt/strategy/searcher/impl/sfp2/StrategyFromPaper2.java

```diff
@@ -0,0 +1,271 @@
+package com.treasure.hunt.strategy.searcher.impl.sfp2;
+
+import com.treasure.hunt.geom.GeometryAngle;
+import com.treasure.hunt.strategy.geom.GeometryItem;
+import com.treasure.hunt.strategy.geom.GeometryType;
+import com.treasure.hunt.strategy.hint.impl.AngleHint;
+import com.treasure.hunt.strategy.searcher.Movement;
+import com.treasure.hunt.strategy.searcher.Searcher;
+import com.treasure.hunt.strategy.searcher.impl.StrategyFromPaper;
+import com.treasure.hunt.utils.JTSUtils;
+import org.locationtech.jts.algorithm.Angle;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.impl.CoordinateArraySequence;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * I tried here to implement the TreasureHunt2 Algorithm,
+ * which for angles beta < 2 * PI has traveling time in O(n ^ (2 - epsilon)).
```

Comment:
```suggestion
 * which for angles beta &lt; 2 * PI has traveling time in O(n ^ (2 - epsilon)).
```
We have to format the javadoc comments in proper HTML for the CI javadoc compilation to work.